com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/StepsUtil.java

```diff
@@ -406,6 +407,36 @@ public static void setAppsToDeploy(DelegateExecution context, List<CloudApplicat
             .collect(Collectors.toList());
         context.setVariable(Constants.VAR_APPS_TO_DEPLOY, cloudApplicationsAsStrings);
     }
+    
+    @SuppressWarnings("unchecked")
+    public static List<CloudApplicationExtended> getIteratedAppsInParallel(DelegateExecution context) {
+        Object appsDeployedInParallel = context.getVariable(Constants.VAR_ITERATED_APPS_IN_PARALLEL);
+        if (appsDeployedInParallel == null) {
+            return new ArrayList<>();
+        }
+        List<String> cloudApplicationsAsStrings = (List<String>) appsDeployedInParallel;
+        return cloudApplicationsAsStrings.stream()
+            .map(app -> (CloudApplicationExtended) JsonUtil.fromJson(app, CloudApplicationExtended.class))

```

Comment:
Why do you need the cast? Isn't JsonUtil.fromJson casting the result to the type of the "class" object? If you don't need it, please use a method reference instead of a lambda.

Reply:
No, I need to explicitly assign it, fromJson ensures that the string will be deserialized to CloudApplicationExtended, but the return type is not the same. Assignment is needed. Check out the method.